asyn/drvAsynSerial/drvAsynIPServerPort.c

```diff
@@ -395,6 +397,16 @@ int createServerSocket(ttyController_t *tty) {
         serverAddr.sin_family = AF_INET;
         serverAddr.sin_addr.s_addr = INADDR_ANY;
         serverAddr.sin_port = htons(tty->portNumber);
+
+        /* got host or interface name, resolve it */
+        if (tty->interfaceInfo != NULL && *tty->interfaceInfo)
+            if (hostToIPAddr(tty->interfaceInfo, &serverAddr.sin_addr) < 0)
+                serverAddr.sin_addr.s_addr = INADDR_ANY;
+
```

Comment:
There are 2 possible improvements:
- Keep "localhost:4002" as is to be backwards compatible
- Fail if the interfaceInfo can not be resolved.
```

        /*
         * got host or interface name, resolve it
         * unless we see "localhost":
         * This may be used be used in existing start scripts
         * and we don't want to break them
         */
        if (tty->interfaceInfo != NULL && *tty->interfaceInfo &&
            strcmp(tty->interfaceInfo, "localhost")) {
            if (hostToIPAddr(tty->interfaceInfo, &serverAddr.sin_addr) < 0) {
                //serverAddr.sin_addr.s_addr = INADDR_ANY;
                printf("Can't lookup '%s' : %s",
                       tty->interfaceInfo, strerror(SOCKERRNO));
                return -1;
            }
        }

        if (ipAddrToDottedIP(&serverAddr, &srvaddrtxt[0], sizeof(srvaddrtxt)) > 0) {
            srvaddrtxt[sizeof(srvaddrtxt) - 1] = '\0';
            printf("serverAddr: %s\n", srvaddrtxt);
        }

```

Reply:
From my point of view, it should be possible to bind only to the loopback interface. A special name as exception could produce confusion.